Docs: Fixes link to ubuntu docker notes

mkaz opened the pull request:
## Description

The env-ubuntu.md file in contributors directory was not included in the toc.json, so it was never published. This PR moves the document to the `devenv/` directory as a good place to collect docs related to setting up a developer environment.

The PR also includes a few minor text tweaks for clarify.

## How has this been tested?

Confirm link to "extended instructions...." for setting up Docker in Ubuntu works as expected.

[View rendered on branch here](https://github.com/WordPress/gutenberg/blob/update/docs-ubuntu-docker/docs/contributors/getting-started.md).

## Types of changes

Documentation.

mkaz committed:
Fix link, move ubuntu docker document to devenv

mkaz committed:
Fix Ubuntu 20.04.1 instructions, fix export var

mkaz commented:
@Aljullu You helped contribute to the Ubuntu and Docker documentation previously, can you help confirm the updated instructions for Ubuntu 20.04.1. I've tested and ran through the binaries included from the default repository and everything worked for me.

Thanks!

Aljullu reviewed (approved):
Thanks for tagging me @mkaz, I tested with a fresh install of Ubuntu 20.04.1 and everything worked as expected. I left some inline comments mostly focused on styling, feel free to ignore them if I was too nitpicking.

mkaz committed:
Apply suggestions from code review

Co-authored-by: Albert Juhé Lluveras <contact@albertjuhe.com>

mkaz committed:
Remove docker-compose instructions

Docker-compose is not needed for most now that Ubuntu binaries are
updated. Just include link to Docker docs which does a better job of
documenting installation.

mkaz commented on docs/designers-developers/developers/tutorials/devenv/docker-ubuntu.md:
```diff
@@ -1,12 +1,18 @@
-After downloading the binary file `docker-compose-Linux-x86_64`, rename to just `docker-compose` and copy it to `/usr/local/bin` or another spot in your PATH.
+	-Follow these [directions from Docker to install](https://docs.docker.com/install/linux/docker-ce/ubuntu/)
+
+	- Additionally, you need to install `docker-compose`, you can follow the [directions from Docker](https://docs.docker.com/compose/install/) or simply [download the latest binary](https://github.com/docker/compose/releases) from GitHub releases.
```
I've removed the explicit directions for docker-compose, if needed (less likely now) the Docker site documents it better.

mkaz commented on docs/designers-developers/developers/tutorials/devenv/docker-ubuntu.md:
```diff
@@ -1,12 +1,18 @@
+
+	- Additionally, you need to install `docker-compose`, you can follow the [directions from Docker](https://docs.docker.com/compose/install/) or simply [download the latest binary](https://github.com/docker/compose/releases) from GitHub releases.
+
+	- After downloading the binary file `docker-compose-Linux-x86_64`, rename to just `docker-compose` and copy it to `/usr/local/bin` or another spot in your PATH.
```
Agreed, it was confusing. I removed it with the docker-compose part. Docker documentation is sufficient.
